vignettes/basketball_ptr/lib/game_score.dart

```diff
@@ -37,7 +39,12 @@ class _GameScoreState extends State<GameScore> with SingleTickerProviderStateMix
     _controller.addStatusListener((status) {
       if (status == AnimationStatus.completed) {
         setState(() {
-          _data = _newData;
+          //TODO: I added this condition because _data's null safety was never checked 
+          //so I concluded that it must not be null. change if needed
+          if (_newData != null) {
+            _data = _newData!;
+          }
```

Comment:
Seems good to me! Just remove the comment? Might as well move the `_newData = null;` inside the `if` too?